### PR TITLE
enable admins to edit dataset default configurations

### LINKED ISSUE #2457: enable admins to edit dataset default configurations
route to get default configuration, and if there is none, get default, with layer list, etc.
+ update routes

<a href="https://timer.scm.io/repos/537237ce1a00001a003c4a14/issues/2457/create?referer=github" target="_blank">Log Time</a>

### PR DESCRIPTION
**Pull request by fm3**
### Mailable description of changes:
 - Admins can now edit dataset default configurations

### Steps to test:
- Click to edit a dataset, change some settings (e.g. position and zoom), save
- Open the dataset in view mode
- Position and Zoom should be preset
- Initial dataset import should still work (w/o configuration options)
- new api routes GET and POST `/api/dataSetConfigurations/default/:dataSetName`

### Issues:
- fixes #2457

------
- [x] Ready for review

### CODE REVIEW

File: app/models/configuration/DataSetConfiguration.scala [hunk 1]
```diff
@@ -1,24 +1,40 @@
...
-    Map(
-      "fourBit" -> JsBoolean(false),
-      "quality" -> JsNumber(0),
-      "interpolation" -> JsBoolean(true),
-      "segmentationOpacity" -> JsNumber(20)))
+  def constructInitialDefault(dataSet: DataSet): DataSetConfiguration =
+    constructInitialDefault(dataSet.dataSource.toUsable.map(d => d.dataLayers.map(_.name)).getOrElse(List()))
+
+  def constructInitialDefault(layerNames: List[String]): DataSetConfiguration = {
+    val layerValues = Json.toJson(layerNames.map(layerName => Json.obj(layerName -> initialDefaultPerLayer)))
```
Comment (normanrz) [inline app/models/configuration/DataSetConfiguration.scala]: @fm3 The layer format is off. It needs to be one map with layer names as keys, not an array of layers.

**Comment by fm3** [comment]
@rschwanhold could you review the backend parts?

**Comment by fm3** [comment]
@rschwanhold that’s right, it is displayed as a label in the Dataset Gallery (I think), which is why it should be sent to the frontend with the dataset json, and be updateable

**Comment by daniel-wer** [comment]
@fm3 Ideally layer configurations would only be configurable (and present) for non-segmentation layers, as the settings won't have any effect for segmentation layers anyway. Is it possible to exclude segmentation layers when you are generating the default layer settings in the backend?

**Comment by fm3** [comment]
I noticed a bug: I tried to import ROI2017_wkw with no prior datasource-properties.json, but the import view crashed with `can’t convert null to object`.

**Pull request merged**